extensions-core/stats/src/main/java/io/druid/query/aggregation/variance/VarianceAggregator.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.variance;
+
+import io.druid.query.aggregation.Aggregator;
+import io.druid.segment.FloatColumnSelector;
+import io.druid.segment.LongColumnSelector;
+import io.druid.segment.ObjectColumnSelector;
+
+/**
+ */
+public abstract class VarianceAggregator implements Aggregator
+{
+  protected final String name;
+
+  protected final VarianceHolder holder = new VarianceHolder();
+
+  public VarianceAggregator(String name)
+  {
+    this.name = name;
+  }
+
+  @Override
+  public void reset()
+  {
+    holder.reset();
+  }
+
+  @Override
+  public Object get()
+  {
+    return holder;
+  }
+
+  @Override
+  public String getName()
+  {
+    return name;
+  }
+
+  @Override
+  public void close()
+  {
+  }
+
+  @Override
+  public float getFloat()
+  {
+    throw new UnsupportedOperationException("VarianceAggregator does not support getFloat()");
+  }
+
+  @Override
+  public long getLong()
+  {
+    throw new UnsupportedOperationException("VarianceAggregator does not support getLong()");
+  }
+
+  public static final class FloatVarianceAggregator extends VarianceAggregator
```

Comment:
My point is, why not only have the VarianceAggregator that only supports doubles? We can avoid requiring float and long aggregators, remove 'FloatVarianceAggregator' and 'LongVarianceAggregator', move the logic in 'ObjectVarianceAggregator' directly in 'VarianceAggregator' and also avoid the getMetricClass() function in IncrementalIndex